_spinnaker_install_admin_guides/install-on-aws-halyard-pod.md

```diff
@@ -0,0 +1,668 @@
+the containers may not be available yet. You can either wait
+or check the status of all of the containers using the command for your cloud provider
+(such as `kubectl get pods --namespace spinnaker`).
+
+# Ingress
+
+There are a number of ways to expose Spinnaker, but basically there are these requirements:
+
+* Given a domain name (or IP address) (such as spinnaker.domain.com or 55.55.55.55)
+* You should be able to reach the `spin-deck` service at the root of the domain (http://spinnaker.domain.com or http://55.55.55.55)
+* You should be able to reach the `spin-gate` service at the root of the domain (http://spinnaker.domain.com/api/v1 or http://55.55.55.55/api/v1)
+* You can use either http or https, as long as you use the same for both
+* You have to configure Spinnaker to be aware of its endpoints
+
+This section details how to do so with the NGINX ingress controller.
+
+## Install the NGINX ingress controller
+
+In order to expose Spinnaker to end users, you have perform the following actions:
+
+- Expose the spin-deck (UI) Kubernetes service on some URL endpoint
+- Expose the spin-gate (API) Kubernetes service on some URL endpoint
+- Update Spinnaker (via Halyard) to be aware of the new endpoints
+
+**If you already have an ingress controller, use that ingress controller instead.  You can check for the existence of the Nginx Ingress Controller by running `kubectl get ns` and looking for a namespace called `ingress-nginx`; if it exists, you likely already have an Nginx Ingress Controller running in your cluster**
+
+We're going to install the NGINX ingress controller on AWS (this uses the Layer 4 ELB, as indicated in the NGINX ingress controller [documentation](https://github.com/kubernetes/ingress-nginx/blob/master/docs/deploy/index.md#aws) - you can use other NGINX ingress controller configurations such as the Layer 7 load balancer per your organization's ingress policy.)
+
+(Both of these are configurable with Spinnaker, but the NGINX ingress controller is also generally much more configurable)
+
+From the `workstation machine` (where `kubectl` is installed):
+
+Install the NGINX ingress controller components:
+
+```bash
+kubectl --kubeconfig kubeconfig-aws apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/mandatory.yaml
+```
+
+Install the NGINX ingress controller AWS-specific service:
+
+```bash
+kubectl --kubeconfig kubeconfig-aws apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/aws/service-l4.yaml
+kubectl --kubeconfig kubeconfig-aws apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/aws/patch-configmap-l4.yaml
+```
+
+## Set up an Ingress for `spin-deck` and `spin-gate`
+
+Get the external IP for the NGINX ingress controller:
+
+```
+kubectl get svc -n ingress-nginx
+```
+
+You'll get a DNS name or IP address in the `EXTERNAL-IP` field.
```

Comment:
Should we add in a quick comment about if you get an error.  like if you can't get an EXTERNAL-IP?  is that possible.  will the NGINX ingress controller work behind a private VPC?  prob not right?  and in which case do we need to tag it?  i'm getting out of my understanding...

Reply:
maybe in v2 (once i figure out the collapsing)